amazon-cloudwatch-container-insights/cloudwatch-agent-dockerfile/source/Dockerfile

```diff
@@ -6,12 +6,16 @@ FROM $GO_IMAGE as builder
 # NOTE: This arg will be populated by docker buildx
 # https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope
 ARG TARGETARCH
+ENV GOPROXY=direct
 RUN mkdir -p /go/src/github.com/aws/amazon-cloudwatch-agent/
 WORKDIR /go/src/github.com/aws/amazon-cloudwatch-agent/
 
 ARG GO111MODULE="on"
 ENV GO111MODULE=${GO111MODULE}
 
+COPY go.mod /go/src/github.com/aws/amazon-cloudwatch-agent/
+COPY go.sum /go/src/github.com/aws/amazon-cloudwatch-agent/
+RUN go mod download -x
```

Comment:
Based on my experience, `go mod download` would download more dependencies than actually needed for the build. However, in step [make build-for-docker](https://github.com/aws/amazon-cloudwatch-agent/pull/437/files#diff-45b77cac272377144caa2b7f83dd47cbc6b66943a7f7049879fd6eec4a791ff3R20), there would be a `go build` which build enough dependencies for the build. Therefore, would be better if we delete `go mod download -x`.